src/ophys_etl/transforms/sine_dewarp.py

```diff
@@ -0,0 +1,606 @@
+        dewarped_file.create_dataset(
+            output_dataset, shape=movie_shape, dtype=movie_dtype
+        )
+    else:
+        out_shape = [
+            movie_shape[0],
+            movie_shape[1],
+            512 - xtable['R'] - xtable['L']
+        ]
+        dewarped_file.create_dataset(
+            output_dataset, shape=out_shape, dtype=movie_dtype
+        )
+    dewarped_file.close()
+
+
+def run_dewarping(FOVwidth: int,
+                  noise_reduction: int,
+                  threads: int,
+                  input_file: str,
+                  input_dataset: str,
+                  output_file: str,
+                  output_dataset: str,
+                  aL: float,
+                  aR: float,
+                  bL: float,
+                  bR: float):
+    """
+    Dewarp the movie frame by frame, utilizing multiprocessing (along with
+    manual chunking of the movie beforehand) to increase efficiency.
+
+    Parameters
+    ----------
+    FOVwidth: int
+        Field of View width. Will be the width of the output image, unless 0,
+        then it will be the width of the input image.
+    noise_reduction: int
+        The noise reduction method that will be used in dewarping.
+    threads: int
+        The number of worker threads to use in multiprocessing.
+    input_file: str
+        File path where the input h5 is saved.
+    input_dataset: str
+        The name of the dataset representing the movie within the
+        input h5 file.
+    output_file: str
+        File path where the h5 containing the dewarped video will be saved.
+    output_dataset: str,
+        The name of the dataset representing the movie within the
+        output h5 file.
+    aL: float
+        How much of the left side of each movie frame has been warped.
+    aR: float
+        How much of the right side of each movie frame has been warped.
+    bL: float
+        Parameter to determine "how warped" the image is on the left side.
+    bR: float
+        Parameter to determine "how warped" the image is on the right side.
+    Returns
+    -------
+    """
+
+    # Get statistics about the movie
+    input_h5_file = h5py.File(input_file, 'r')
+    movie = input_h5_file[input_dataset]
+
+    movie_shape = movie.shape
+    movie_dtype = movie.dtype
+    T, y_orig, x_orig = movie.shape
+
+    # IMPORTANT: estimated modevalue is from 8bit img but convert back to 16bit
+    xtable = create_xtable(movie, aL, aR, bL, bR, noise_reduction)
+
+    make_output_file(output_file, output_dataset, xtable,
+                     FOVwidth, movie_shape, movie_dtype)
+
+    start_time = time.time()
+    with multiprocessing.Pool(threads) as pool, \
+         h5py.File(output_file, "a") as f:
+
+        fn = functools.partial(
+            xdewarp,
+            FOVwidth=FOVwidth,
+            xtable=xtable,
+            noise_reduction=noise_reduction
+        )
+
+        chunk_size = 1000
+        for chunk in range(0, T, chunk_size):
```

Comment:
why not [numpy.array_split](https://numpy.org/doc/stable/reference/generated/numpy.array_split.html#numpy-array-split)?

Reply:
I was worried about that doubling up on the memory usage, since it would still have the full movie in memory, plus the new list of chunks. So I just indexed into the movie array instead.

Do you still think it would be better to use numpy's method?

Reply:
The numpy docs indicate that it provides views of the original array.